core/src/main/java/org/frankframework/pipes/BaseReplacingInputStream.java

```diff
@@ -0,0 +1,83 @@
+/*
+   Copyright 2018-2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.pipes;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import jakarta.annotation.Nonnull;
+
+/**
+ * Since {@link ReplacingVariablesInputStream} and {@link ReplacingInputStream} share some base overrides,
+ * these methods were extracted to this base class.
+ *
+ * @author Erik van Dongen
+ */
+public abstract class BaseReplacingInputStream extends FilterInputStream {
```

Comment:
Ik denk dat deze hele class weg kan en je beter direct InputStream kan extenden.

Reply:
Maar het gaat erom dat `read()` in de subclass wordt gebruikt (zie regel 73). Dat krijg je niet voor elkaar op die manier.

Reply:
Ik ben heel erg mijn best aan het doen om maar ik snap hem toch nog niet helemaal... Zie https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/io/InputStream.java